components/script/dom/htmlimageelement.rs

```diff
@@ -143,9 +197,42 @@ impl HTMLImageElement {
                         UpdateReplacedElement, runnable));
                 });
 
+                /// srm912: This needs to be replaced by the commented code below
+                ///         A new ImageContext needs to be created for the NetworkListener
+                ///         which needs to implement the AsyncResponseListener and PreInvoke
+                ///         traits 
+                ///         OR 
+                ///         Do we need to use the ImageRequest struct that was created earlier
+                ///         since it also has the relevant attributes such as data vector, state and URL
```

Comment:
Using a new ImageContext makes sense for me. We'll probably want to manipulate the ImageRequest members based on the data provided to the ImageContext methods, but the proper way to do that isn't clear to me yet so it's not the goal of this project.